src/graph_1.py

```diff
@@ -35,24 +39,21 @@ def del_node(self, val):
                     temp_edges[key] = self.edges[key]
             self.edges = temp_edges
         else:
-            raise IndexError("Node not in graph.")
+            raise Exception("Node not in graph.")
 
     def del_edge(self, val1, val2):
         """Del edge connecting "val1" and "val2"."""
         if (val1, val2) in self.edges:
             del self.edges[(val1, val2)]
         else:
-            raise IndexError("Edge does not exist.")
+            raise Exception("Edge does not exist.")
 
     def has_node(self, val):
         """Return true of node w/ "val" is in the graph."""
-        if val in self.node_list:
-            return True
-        else:
-            return False
+        return (val in self.node_list)
 
     def neighbors(self, val):
-        """Return list of all nodes connedted to node with "val"."""
+        """Return list of all nodes connected to node with "val"."""
         temp_list = []
```

Comment:
Not docking you points for this, but you could just initialize this as a set and then do `temp_list.add(something)` whenever you want to add something to the set and have it be unique